### ISSUE TITLE
Leave clean version string for tagged release commits only

### ISSUE THREAD

**Issue by dahaic** [OpenSCAP/openscap#639]
I have noticed openscap reports itself as next bugfix version for a whole time this bugfix version is developed. Could we have some suffix added, unless it is really released version? That way, it would be easily discernible, whether I have installed version real `1.2.13` or some commit between `1.2.12` and `1.2.13` release.

Just a simple `1.2.13-alpha` would be sufficient - it's not necessary to distinguish between individual commits.

**Comment by rsprudencio** [comment]
That makes sense, we could borrow -rcX from other projects like linux kernel.

**Comment by mpreisler** [comment]
We could even have a `-git` suffix or something like that in there always and only remove it locally when preparing source tarballs.

**Comment by jan-cerny** [comment]
I have looked into this a little.
To change this, it is enough to change the version string in `ac_probes/configure.ac.tpl` (and `configure.ac`.

But if we would like to do a release, we would have to change the version manually before running make dist.

I tried to find a way how to remove the "-git" suffix when running make dist automatically, but I was unable to find a solution.

There is a way how to compute version number dynamically, by arbitrary script, based eg. on last 
commit id in git. The autoconf itself uses this 😬 
See https://lists.gnu.org/archive/html/autoconf/2007-12/msg00027.html

We could do some script that will remove the "git" suffix from the version string when it founds that this version has been already tagged in git (I mean `git tag`). This would solve problem only for the releases. What about some testing, eg. scratch builds?

Any thoughts?
